reference/tests/AddOpTest.cpp

```diff
@@ -0,0 +1,290 @@
+    }
+  })";
+  runTestCase(kModule, {/*operands*/
+                        {"1", "2", "4294967295", "4"},
+                        {"4294967295", "4294967295", "4294967295", "10"},
+                        /*expected result*/
+                        {"0", "1", "4294967294", "14"}});
+}
+
+TEST(AddOpInterpreterTest, SInt64WithOv) {
+  constexpr llvm::StringRef kModule = R"(
+  module {
+    func.func @main(%arg0: tensor<1x9x1xi64>, %arg1: tensor<1x9x1xi64>) -> tensor<1x9x1xi64> {
+      %result = stablehlo.add %arg0, %arg1 : tensor<1x9x1xi64>
+      func.return %result : tensor<1x9x1xi64>
+    }
+  })";
+  runTestCase(
+      kModule,
+      {/*operands*/
+       {"1", "2", "-9223372036854775808", "-9223372036854775808", "16", "-16",
+        "16", "9223372036854775807", "-9223372036854775808"},
+       {"9223372036854775807", "9223372036854775807", "-1",
+        "-9223372036854775808", "16", "-16", "-16", "0", "0"},
+       /*expected result*/
+       {"-9223372036854775808", "-9223372036854775807", "9223372036854775807",
+        "0", "32", "-32", "0", "9223372036854775807", "-9223372036854775808"}});
+}
+
+TEST(AddOpInterpreterTest, UInt64WithOv) {
+  constexpr llvm::StringRef kModule = R"(
+  module {
+    func.func @main(%arg0: tensor<2x2xui64>, %arg1: tensor<2x2xui64>) -> tensor<2x2xui64> {
+      %result = stablehlo.add %arg0, %arg1 : tensor<2x2xui64>
+      func.return %result : tensor<2x2xui64>
+    }
+  })";
+  runTestCase(kModule, {/*operands*/
+                        {"1", "2", "18446744073709551615", "4"},
+                        {"18446744073709551615", "18446744073709551615",
+                         "18446744073709551615", "10"},
+                        /*expected result*/
+                        {"0", "1", "18446744073709551614", "14"}});
+}
+
+TEST(AddOpInterpreterTest, Complex32) {
+  constexpr llvm::StringRef kModule = R"(
+  module {
+    func.func @main(%arg0: tensor<2xcomplex<f32>>, %arg1: tensor<2xcomplex<f32>>) -> tensor<2xcomplex<f32>> {
+      %result = stablehlo.add %arg0, %arg1 : tensor<2xcomplex<f32>>
+      func.return %result : tensor<2xcomplex<f32>>
+    }
+  })";
+  runTestCase(kModule, {/*operands*/
+                        {"1.5", "2.5", "7.5", "5.5"},
+                        {"1.5", "2.5", "7.5", "5.5"},
+                        /*expected result*/
+                        {"3.0", "5.0", "15.0", "11.0"}});
+}
+
+TEST(AddOpInterpreterTest, Complex64) {
+  constexpr llvm::StringRef kModule = R"(
+  module {
+    func.func @main(%arg0: tensor<2xcomplex<f64>>, %arg1: tensor<2xcomplex<f64>>) -> tensor<2xcomplex<f64>> {
+      %result = stablehlo.add %arg0, %arg1 : tensor<2xcomplex<f64>>
+      func.return %result : tensor<2xcomplex<f64>>
+    }
+  })";
+  runTestCase(kModule, {/*operands*/
+                        {"1.5", "2.5", "7.5", "5.5"},
+                        {"1.5", "2.5", "7.5", "5.5"},
+                        /*expected result*/
+                        {"3.0", "5.0", "15.0", "11.0"}});
+}
```

Comment:
Can we find a way to use a more MLIR friendly way of testing? That is a test bench that fits in lit with .mlir files?

Reply:
I have the following way in mind. 

```
// RUN: stablehlo-interpreter-runner -split-input-file  %s | FileCheck %s

// CHECK-LABEL:       tensor<i32> {
// CHECK-NEXT:          3 : i32
// CHECK-NEXT:        }
func.func @main() -> tensor<i32> {
  %0 = stablehlo.constant dense<1> : tensor<i32>
  %1 = stablehlo.constant dense<2> : tensor<i32>
  %2 = stablehlo.add %0, %1 : tensor<i32>
  func.return %2 : tensor<i32>
}

// --- 

// More tests

```

 - The tool `stablehlo-interpreter-runner` will consume the IR snippet and print the result tensor(s) to be matched uisng the CHECK directives.
 - The input operand values will be provided using `stable.constant` ops. Implementation wise, that includes defining the `eval` function for  `stablehlo.constant` op.
 
What do you think?

Reply:
Removed the gtests and added  the lit tests instead. 

https://github.com/openxla/stablehlo/pull/44/files#diff-0842b6fc7199e52d0740a7b91341e7659fe4007820194b536a2bb57e22b395b3